src/alexapi/playback_handlers/soxhandler.py

```diff
@@ -0,0 +1,101 @@
+import subprocess
+from basehandler import BaseHandler, PlaybackAudioType
+
+
+MAX_VOLUME_GAIN=30
+
+
+class SoxHandler(BaseHandler):
+    def __init__(self, config, callback_report):
+        super(SoxHandler, self).__init__(config, callback_report)
+        self.__config = config
+
+        self.volume_gain = 0
+        self.media_volume_gain = 0
+
+        self.parametersSpeech = []
+        self.parametersMedia = []
+
+    def on_setup(self):
+        if self.__config['sound']['output']:
+            self.parametersSpeech.extend(['-t', self.__config['sound']['output']])
+            if self.__config['sound']['output_device']:
+                self.parametersSpeech.append(self.__config['sound']['output_device'])
+
+        if self.__config['sound']['media_output']:
+            self.parametersMedia.extend(['-t', self.__config['sound']['media_output']])
+            if self.__config['sound']['media_output_device']:
+                self.parametersMedia.append(self.__config['sound']['media_output_device'])
+        else:
+            self.parametersMedia = self.parametersSpeech
+
+        if self.__config['sound']['default_volume']:
+            self.on_set_volume(self.__config['sound']['default_volume'])
+        if self.__config['sound']['media_default_volume']:
+            self.on_set_media_volume(self.__config['sound']['media_default_volume'])
+
+        self.playback_padding = str(self.__config['sound']['playback_padding'])
+
+    def on_play(self, item):
+        # SoX can't play file URLs
+        audio_file = item.url.replace('file://', '')
+
+        sox_cmd = ['sox', '-q', audio_file]
```

Comment:
The `play` and `rec` commands are just shortcuts that delegate back to the main sox code. All of the same functionality is available in the main `sox` executable. Furthermore the output devices that can be changed with the `-t` switch of the `sox` command have to be set with environment variables when using `play` and `rec`. It's simpler to use the sox command directly plus it keeps the code from having to use the `shell=True` attribute of Popen.